great_expectations/data_context/data_context.py

```diff
@@ -1260,6 +1260,11 @@ def _build_datasource_from_config(
         if isinstance(config, DatasourceConfig):
             config = datasourceConfigSchema.dump(config)
         config.update({"name": name})
+        if config["class_name"] in [
```

Comment:
I assume this is here because some types of Datasource need a `data_context_root_directory`, and others don't?

Two thoughts:
1. This would be a good place for a 1- or 2-line clarifying comment.
2. Wouldn't `isinstance` be a better way to guarantee forward compatibility?

Reply:
This question does not need to block merging.

Reply:
@abegong You are correct (and I will add a comment).  I did not want to modify the legacy classes (i.e., to allow them to accept the `data_context_root_directory`), hence ended up checking the `class_name`.

Using `isinstance` would not work here, because we do not have the `datasource` instance, until the *entire* `datasource` hierarchy (i.e., including our deep-layered `DataConnector` structure) is instantiated.  At this point, it is too late, because the `DataConnector` that needed `data_context_root_directory` was instantiated with it being `None`.